File: 강의 자료/02-알고리즘/09~11-그래프 탐색/문제별 코드/11724-연결 요소의 개수/solution.py
import sys
from collections import deque
si = sys.stdin.readline
n, m = list(map(int, si().split()))
adj = [[] for _ in range(n + 1)]

for _ in range(m):
    x, y = list(map(int, si().split()))
    adj[x].append(y)
    adj[y].append(x)

visit = [False] * (n + 1)
def dfs(x):
    visit[x] = True
    for y in adj[x]:
        if visit[y]:
            continue
        dfs(y)

ans = 0
for i in range(1, n + 1):
    if visit[i]: continue
    ans += 1
    dfs(i)

print(ans)

File: 강의 자료/02-알고리즘/09~11-그래프 탐색/문제별 코드/11724-연결 요소의 개수/solution.cpp
#include <iostream>
#include <algorithm>
#pragma warning(disable:4996)
using namespace std;

#include <vector>
vector<int> adj[1005];
int n, m;
void input() {
	cin >> n >> m;
	for (int i = 1; i <= m; i++) {
		int u, v;
		cin >> u >> v;
		adj[u].push_back(v);
		adj[v].push_back(u);
	}
}

bool visit[1005];

void DFS(int x) {
	visit[x] = true;
	for (int y : adj[x]) {
		if (visit[y]) continue;
		DFS(y);
	}
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(0);
	input();
	int ans = 0;
	for (int i = 1; i <= n; i++) {
		if (visit[i]) continue;
		ans++;
		DFS(i);
	}
	cout << ans;
	return 0;
}